File: JS/Programmers/Level1/05 크레인 인형뽑기 게임.js
function solution(board, moves) {
  var answer = 0;

  var basket;
  var spin = 0;

  for (let i = 0; i < moves.length; i++) {
    var index = moves[i] - 1;
    spin = 0;
    for (let j = 0; j < board.length; j++) {
      if (board[j][index] != 0 && spin == 0) {
        if (basket.length != 0) {
          if (board[j][index] == basket[basket.length - 1]) {
            answer = answer + 2;
            basket.pop();
          } else if (board[j][index] != basket[basket.length - 1]) {
            basket.push(board[j][index]);
          }
        } else if (basket.length == 0) {
          basket.push(board[j][index]);
        }
        board[j][index] = 0;
        spin = 1;
      }
    }
  }
  return answer;
}

File: C++/Programmers/Level1/05 크레인 인형뽑기 게임.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> board, vector<int> moves) {
    int answer = 0;
    
    vector<int> basket;
    int spin = 0;

    for(int i=0; i<moves.size(); i++){
        int index = moves[i]-1;
        spin = 0;
        for(int j=0; j<board.size(); j++){
            if(board[j][index]!=0 && spin == 0){
                if(!basket.empty()){
                    if(board[j][index] == basket.back()){
                        answer = answer + 2;
                        basket.pop_back();
                    }
                    else if (board[j][index] != basket.back()){
                        basket.push_back(board[j][index]);
                    }
                }
                else if (basket.empty()){
                    basket.push_back(board[j][index]);
                }
            board[j][index] = 0;
            spin = 1;
            }
        }
    }

    return answer;
}